reo/validators.py

```diff
@@ -1881,6 +1882,43 @@ def check_special_cases(self, object_name_path, template_values=None, real_value
                         else:
                             updated_set[param] = real_values.get(param)
 
+        if object_name_path[-1] == "GHP":
+            if self.isValid:
+                # Look up default GHP effciency thermal factors if not input
+                eval_ghp = False
+                if real_values.get("building_sqft") is not None:
+                    eval_ghp = True
+                if eval_ghp:
+                    heating_factor = real_values.get("space_heating_efficiency_thermal_factor")
+                    cooling_factor = real_values.get("cooling_efficiency_thermal_factor")
+                    if heating_factor in [[], None] or cooling_factor in [[], None]:
+                        latitude = self.input_dict['Scenario']['Site']['latitude']
+                        longitude = self.input_dict['Scenario']['Site']['longitude']
+                        climate_zone, nearest_city, geometric_flag = get_climate_zone_and_nearest_city(latitude, longitude, BuiltInProfile.default_cities)
+                        heating_factor_data = pd.read_csv(os.path.join('input_files', 'LoadProfiles', 'ghp_heating_efficiency_thermal_factors.csv'), index_col="Building Type")
+                        cooling_factor_data = pd.read_csv(os.path.join('input_files', 'LoadProfiles', 'ghp_cooling_efficiency_thermal_factors.csv'), index_col="Building Type")
+                        building_type_heating = self.input_dict['Scenario']['Site']['LoadProfileBoilerFuel'].get('doe_reference_name') or []
+                        building_type_cooling = self.input_dict['Scenario']['Site']['LoadProfileChillerThermal'].get('doe_reference_name') or []
+
+                    # Default thermal factors are assigned for certain building types and not for campuses (multiple buildings)
```

Comment:
would it be too much to define the heating factor for a group of buildings as a weighted sum based on the annual_kwh? I would think that GHP systems for groups of buildings might be a common use case.

Reply:
@NLaws That is a good idea, and I plan to do that, but I'd like to leave that as a future addition due to time constraints here with Andrew then having to interface with the endpoint for getting these defaults based on multiple reference buildings. I'd also want to run that buy the buildings people. For now, the users can enter their own custom thermal factors for campuses.